README.md

```diff
@@ -196,10 +212,245 @@ The following syntax shall be used to declare an array:
+        datatype: string
+        range: $ == "hello" or $ == "world"
+```
+
+
+If the range expression, with `$` substituted by the value to verify, evaluates to true, then the value is legal.
+
+
+## Range specification for enumerations
+
+Parameters, errors, and properties that have an enum value specify
+their range using the the symbolic values, as shown below:
+
+
+```YAML
+enumerations:
+  - name: error_t
+    type: enumeration
+
+    datatype: int16
+    options:
+      - name: "null"
+        value: 0
+
+      - name: ok
+        value: 1
+
+      - name: in_progress
+        value: 2
+
+methods:
+  - name: my_method
+    error:
+      datatype: .stdvsc.error_t
+      # Symbolic enum value
+      range: $ == "ok" or $ == "in_progress"
+```
+
+
+## Range specification for complex types
+
+Complex (struct) values specify their ranges using member specification after the `$` token, as shown below
+
+```YAML
+structs:
+  - name: seat_location_t
+    members:
+      - name: row
+        datatype: uint8
+      - name: index
+        datatype: uint8
+
+methods:
+  - name: move
+    in:
+      - name: location
+        datatype: seat_location_t
+        range:  $.row >= 0 and $.row <= 3 and $.index in_interval(1, 10)
+```
+
+Nested complex types can be specified as well by using nested members:
+
+```YAML
+structs:
+  - name: inner_struct_t
+    members:
+      - name: inner_struct_value_1
+        datatype: int8
+
+      - name: inner_struct_value_2
+        datatype: int32
+
+  - name: outer_struct_t
+    members:
+      - name: outer_struct_value_1
+        datatype: string
+
+      - name: outer_struct_value_2
+        datatype: float
+
+      - name: an_inner_struct
+        datatype: innert_struct_t
+
+methods:
+  - name: my_function
+    in:
+      - name: my_complex_parameter
+        datatype: outer_struct
+        range: >
+            $.outer_struct_value_1 in_set("first", "second", "fourth", "eigth") and 
+            $.outer_struct_value_2 < 2.0 and
+            ( $.an_inner_struct.inner_struct_value_1 == 1 or
+              $.an_inner_struct.inner_struct_value_2 == 123456789 )
+```
+
+
+## Range specifications for arrays
+
+Range specifications for arrays are defined by adding an index
+(starting with `0`) to the `$` token, as shown below:
+
+
+```YAML
+methods:
+  - name: my_method
+    in:
+      - name: an_array_argument
+        datatype: int8[]
+        arraysize: 3
+        range: $[0] == 10 and $[1] == 20 and $[2] == 30
+```
+
+A range expression that uses an element index that does not exist in
+the tested value will render the expression false. This is especially
+important for unbound arrays where the value can host any number of
+elements.
+
+
+If the value is of a complex type where the type's members are arrays,
+a combination of member names and array indices can be used, as shown
+below:
+
+
+```YAML
+structs:
+  - name: struct_with_array_members_t
+    members:
+      - name: array_member_1[]
```

Comment:
As discussed in the meeting move `[]` down one line

Reply:
I included this fix also